app/src/main/java/org/sil/hearthis/RecordActivity.java

```diff
@@ -34,42 +33,52 @@
 import android.widget.Button;
 import android.widget.ScrollView;
 import android.widget.TextView;
+import android.widget.Toast;
 
-public class RecordActivity extends Activity {
-	
-	int _activeLine;
-	ViewGroup _linesView;
-	int _lineCount;
-    int _bookNum;
-    int _chapNum;
-    IScriptProvider _provider;
+public class RecordActivity extends ActionBarActivity
+{
+	ViewGroup linesView;
+	Button record;
+	Menu recordMenu;
+
+	int activeLine = 0;
+	int lineCount;
+	int chapter;
+	BookInfo book;
+	IScriptProvider provider;
+	int backColour = 0xffffff;
+
+	String fileName = "";
 	
 	Typeface mtfl;
 	
 	MediaRecorder recorder = null;
 	
 	@Override
-	protected void onCreate(Bundle savedInstanceState) {
-		super.onCreate(savedInstanceState);
+	protected void onCreate(Bundle state)
+	{
+		super.onCreate(state);
 		setContentView(R.layout.activity_record);
 		
 		mtfl = (Typeface)Graphite.addFontResource(getAssets(), "CharisSILAfr-R.ttf", "charis", 0, "", "");
-		
+
+		if (state != null)
+			activeLine = state.getInt("activeLine");
```

Comment:
It's a good idea to keep track of the active line. Note that we will need to reset it when advancing to the next chapter.